plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -115,7 +118,15 @@ public PinotTableHandle getTableHandle(ConnectorSession session, SchemaTableName
             DynamicTable dynamicTable = DynamicTableBuilder.buildFromPql(this, tableName);
             return new PinotTableHandle(tableName.getSchemaName(), dynamicTable.getTableName(), TupleDomain.all(), OptionalLong.empty(), Optional.of(dynamicTable));
         }
-        return new PinotTableHandle(tableName.getSchemaName(), tableName.getTableName());
+
+        try {
+            String pinotTableName = getPinotTableNameFromTrinoTableName(tableName.getTableName());
```

Comment:
This seems to also fix a possible bug where if Pinot already has a table with a different case Trino might've created a new one and then would fail to query them.

Reply:
Can we add a test where `test_table` already exists and we try to create `TeSt_TaBlE`. It should fail.

Reply:
Even if we try "create TeSt_TaBlE" in Trino, `PinotMetadata.getTableHandle()` recognizes as a lower case table name because of `SchemaTableName` argument. 

I would add another case to TestPinotMetadata as https://github.com/trinodb/trino/pull/8859/files#diff-23d04803adfe20ed595663294bb3803f43828b85c667b119fa2492e49fb78664R51-R52 instead. This test will fail if `PinotMetadata.getTableHandle()` couldn't look up a table name having uppercase. 

Another possible test is preparing an upper case table in Pinot side and verify CT failure with lower case in Trino. But, I assume added test in TestPinotMetadata is sufficient at this time.

Reply:
> preparing an upper case table in Pinot side and verify CT failure with lower case in Trino

This was my intention. But the new test you added looks sufficient. As long metadata returns lowercase name always engine will take care of the rest.